src/Native/Bootstrap/main.cpp

```diff
@@ -292,16 +292,70 @@ static const pfn c_classlibFunctions[] = {
 
 #endif // !CPPCODEGEN
 
+static bool RUNTIME_INITIALIZED = false;
 extern "C" void InitializeModules(void* osModule, void ** modules, int count, void ** pClasslibFunctions, int nClasslibFunctions);
+extern "C" int InitializeRuntime();
 
+#ifdef CORERT_DLL
 #if defined(_WIN32)
+#define wmain CoreRT_wmain
```

Comment:
These redefinitions of wmain/main should not be needed now. The whole `wmain/main` can be under one big `#ifndef CORERT_DLL` - it should not be needed for CORERT dll.